### PR TITLE
Ci/better ci

### PR DESCRIPTION
**Pull request by lpascal-ledger**

### CODE REVIEW

**Review by cseguret-ledger — COMMENTED** [review]
Good but check comment

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
 on:
+  workflow_dispatch:
   push:
     branches:
       - master
   pull_request:
     branches:
       - master
+      - develop
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply .github/workflows/ci-workflow.yml:11]: Indeed, both syntax are valid. I do not have opinion about which is better, aestheticly speaking.

File: .github/workflows/lint-workflow.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@ on:
   pull_request:
     branches:
     - master
+    - develop
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply .github/workflows/lint-workflow.yml:10]: Some response

File: .github/workflows/ci-workflow.yml [hunk 3]
```diff
@@ -113,28 +134,30 @@ jobs:
           echo $DISPLAY
       - name: Checkout
         uses: actions/checkout@v2
+
       - run: sudo apt-get update -y && sudo apt-get install -y libusb-1.0.0 libudev-dev
+
       - name: Install node
         uses: actions/setup-node@v2
         with:
           node-version: "14.4.0"
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply .github/workflows/ci-workflow.yml:143]: Well, this PR is about the structure of the CI, and enforcing existing tests. I do not know which version of node the dev considers to be the reference, so I didn't touch it.

File: .github/workflows/ci-workflow.yml [hunk 4]
```diff
@@ -113,28 +134,30 @@ jobs:
...
       - name: Gather elfs
-        run: |
-          cd tests/elfs
-          cp `find . -name "*.elf"` .
+        run: cp `find tmp/e2e_elfs/ -name "*.elf"` tests/elfs/
+
       - name: Run zemu tests
-        run: |
-          cd tests && yarn test
+        run: cd tests && yarn test
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply .github/workflows/ci-workflow.yml:163]: Good point, and it also manages the deps install with `make  install_tests`. Fixing it.

[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply .github/workflows/ci-workflow.yml]: Actually, I'm not going to use the `Makefile`. It is too much of a burden: lots of checks are performed before the target is even reached, needing `BOLOS_SDK` variables to be set, including other `Makefile`s, ... it need a proper compilation environment. But even then, some checks will fail if I use `make install_test` then `make run_test` as a `git status` will trigger an error.
We don't need all this hassle here, or using convoluted confs to bypass these checks. ` cd tests && yarn test` will do.

**Review by cseguret-ledger — APPROVED** [review]
LGFM

**Pull request merged**